src/Microsoft.OpenApi.OData.Reader/Operation/NavigationPropertyOperationHandler.cs

```diff
@@ -55,7 +60,8 @@ protected override void Initialize(ODataContext context, ODataPath path)
             NavigationSource = navigationSourceSegment.NavigationSource;
 
             LastSegmentIsKeySegment = path.LastSegment is ODataKeySegment;
-            LastSegmentIsRefSegment = path.LastSegment is ODataRefSegment;
+            LastSegmentIsRefSegment = path.LastSegment is ODataRefSegment;
+            SecondLastSegmentIsKeySegment = Path.Segments.Reverse().Skip(1).Take(1).Single().Kind == ODataSegmentKind.Key;
```

Comment:
should we check that we have at least 2 elements in the segments collection before calling skip/take/single?

Reply:
We can, but I don't think we'd ever have such a scenario where we have less than 2 segments, since `/$ref` segments will always be appended to at least: 
`~navSource/col-nav-prop/{id}/$ref` or `~navSource/nav-prop/$ref`